marlowe-dashboard-client/src/Welcome/Types.purs

```diff
@@ -2,16 +2,21 @@ module Welcome.Types
   ( State
   , Card(..)
   , Action(..)
+  , WalletNicknameOrIdError(..)
   ) where
 
 import Prelude
 import Analytics (class IsEvent, defaultEvent, toEvent)
+import Clipboard (Action) as Clipboard
 import Data.Maybe (Maybe(..))
 import InputField.Types (Action, State) as InputField
+import InputField.Types (class InputFieldError)
 import Types (WebData)
-import WalletData.Types (WalletDetails, WalletLibrary, WalletNickname)
-import WalletData.Validation (WalletIdError, WalletNicknameError, WalletNicknameOrIdError)
+import WalletData.Types (WalletDetails, WalletIdError, WalletLibrary, WalletNickname, WalletNicknameError)
 
+-- TODO (possibly): The WalletData submodule used in the Dashboard has some properties and
+-- functionality that's similar to some of what goes on here. It might be worth generalising it so
+-- it works in both cases, and including it as a submodule here too.
```

Comment:
which part? the card?

Reply:
Not the view so much, but the state. In both cases we have a `WalletLibrary` map and we need to be able to lookup and insert `WalletData` to that. And we use the same two `InputFields` (one for the nickname and one for the wallet ID) for inserting. Enough of the logic is different that I'm not convinced it's worth generalising... but enough of it is the same that's it's worth considering. :shrug: